##AIM
<ul>
<li>To write a program that implements the decision tree for a sample training data set stored</li>
<li>Compute
the accuracy of the classifier, considering few test data sets
</li>

</ul>

##Given
<ul>
<li>Car Evaluation Dataset</li>
<li>Classification of four classes unaccounted, accounted , good , very good.
<li>Algorithm Used is Decision Tree classifier algorithm</li>
</ul>


Dataset link : https://drive.google.com/drive/folders/1Yzil29iBhAOZlueZcq2bgBLtlxMXSOdF?usp=sharing

##Importing Libraries

In [35]:
import numpy as np 
import pandas as pd 

##Data preprocessing

In [36]:
df = pd.read_csv("car_evaluation.csv", header=None)

In [56]:
df.info

<bound method DataFrame.info of      buying  maint  doors persons lug_boot safety  class
0     vhigh  vhigh      2       2    small    low  unacc
1     vhigh  vhigh      2       2    small    med  unacc
2     vhigh  vhigh      2       2    small   high  unacc
3     vhigh  vhigh      2       2      med    low  unacc
4     vhigh  vhigh      2       2      med    med  unacc
...     ...    ...    ...     ...      ...    ...    ...
1723    low    low  5more    more      med    med   good
1724    low    low  5more    more      med   high  vgood
1725    low    low  5more    more      big    low  unacc
1726    low    low  5more    more      big    med   good
1727    low    low  5more    more      big   high  vgood

[1728 rows x 7 columns]>

In [58]:
df.shape

(1728, 7)

In [37]:
col_names = ['buying', 'maint', 'doors', 'persons', 'lug_boot', 'safety', 'class']
df.columns = col_names
df.head()

buying  maint doors persons lug_boot safety  class
0  vhigh  vhigh     2       2    small    low  unacc
1  vhigh  vhigh     2       2    small    med  unacc
2  vhigh  vhigh     2       2    small   high  unacc
3  vhigh  vhigh     2       2      med    low  unacc
4  vhigh  vhigh     2       2      med    med  unacc

In [38]:
X = df.drop(['class'], axis=1)
y = df['class']
#Dividing into dependent and independent variables
#Here Y is the dependent variable and X is the Independent Variable

##Splitting into test and training set

In [48]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.33, random_state = 42)

In [ ]:
pip install category_encoders


##To Change the catogerical Data into num values

In [49]:
import category_encoders as ce
encoder = ce.OrdinalEncoder(cols=['buying', 'maint', 'doors', 'persons', 'lug_boot', 'safety'])
X_train = encoder.fit_transform(X_train)
X_test = encoder.transform(X_test)


In [42]:
X_train.head()

buying  maint  doors  persons  lug_boot  safety
48         1      1      1        1         1       1
468        2      1      1        2         2       1
155        1      2      1        1         2       2
1721       3      3      2        1         2       2
1208       4      3      3        1         2       2

In [61]:
X_train.shape

(1157, 6)

In [62]:
X_test.shape

(571, 6)

##Feature Scaling

In [50]:
from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)

##Using Decision tree Classifier

In [51]:
from sklearn.tree import DecisionTreeClassifier
classifier = DecisionTreeClassifier(criterion = 'entropy', random_state = 0)
classifier.fit(X_train, y_train)

DecisionTreeClassifier(criterion='entropy', random_state=0)

##Test Set

In [52]:
y_pred = classifier.predict(X_test)


##Predicting the Accuracy score

In [60]:
from sklearn.metrics import  accuracy_score
ascore = accuracy_score(y_test, y_pred)
print(f"Score for this model : {100*ascore:.2f} % accuracy\n")

Score for this model : 94.75 % accuracy



#Doing a Single Prediction

In [54]:
print(classifier.predict(sc.transform([[2,3,3,2,2,1]])))
#According to these input values of the features the given prediction was unaccounted

['unacc']


/usr/local/lib/python3.7/dist-packages/sklearn/base.py:446: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names
  "X does not have valid feature names, but"


##Result

<ul>
<li>The accuracy score for this model was 94.75 percentage</li>
<li>Therefore the implementation of Decision Tree Classification and the accuracy score was computed</li>
</ul>